cumulusci/cli/cci.py

```diff
@@ -854,11 +854,58 @@ def service_info(runtime, service_name, plain):
 # Commands for group: org
 
 
+def set_org_name(required):
+    """Generate a callback that enforces the 'required' rule as required"""
+    # could be generalized to work for any mutex pair (or list) but no obvious need
+    def callback(ctx, param, value):
+        """Callback which enforces mutex and 'required' behaviour (if required)."""
+        prev_value = ctx.params.get("org_name")
+        if value and prev_value and prev_value != value:
+            raise click.UsageError(
+                f"Either ORGNAME or --org ORGNAME should be supplied, not both ({value}, {prev_value})"
+            )
+        ctx.params.setdefault("org_name", value)
```

Comment:
```suggestion
        else:
            ctx.params["org_name"] = value
```

(Correct me if I'm overlooking some semantic difference, but if not I'd rather write it this way so the reader doesn't need to be familiar with setdefault)